src/main/java/at/hannibal2/skyhanni/features/chat/SoundResponse.kt

```diff
@@ -13,23 +14,39 @@ object SoundResponse {
 
     private val config get() = SkyHanniMod.feature.chat.soundResponse
 
-    private val repoGroup = RepoPattern.group("chat.sound.response")
-
-    /** REGEX-TEST: meow
-    REGEX-TEST:  meow
-    REGEX-TEST:  meow
-    REGEX-TEST: MEow
-    REGEX-TEST: §ameow
-    REGEX-TEST: hello §ameow
-     */
-    private val meow by repoGroup.pattern("meow", "(?:^|^.* )(?: |§.)*(?i)meow(?: |§.)*(?:\$| .*\$)")
-    private val bark by repoGroup.pattern("bark", "(?:^|^.* )(?: |§.)*(?i)(?:bark|arf|woof)(?: |§.)*(?:\$| .*\$)")
+    init {
+        SoundResponseTypes.entries.forEach { it.pattern }
+    }
 
     @SubscribeEvent
     fun onLorenzChat(event: LorenzChatEvent) {
-        when {
-            config.meow && meow.matches(event.message) -> SoundUtils.playMeowSound()
-            config.bark && bark.matches(event.message) -> SoundUtils.playBarkSound()
+        for (soundType in SoundResponseTypes.entries) {
+            if (!config.soundResponses.contains(soundType)) continue
+            if (soundType.pattern.matches(event.message)) {
+                soundType.sound.playSound()
+                return
+            }
         }
     }
 }
+
+private const val START_PATTERN = "(?:^|^.* )(?: |§.)*(?i)"
+private const val END_PATTERN = "(?: |§.|!|\\?|\\.)*(?:\$| .*\$)"
+
+enum class SoundResponseTypes(soundLocation: String, triggersOn: List<String>) {
+    CAT("mob.cat.meow", listOf("meow")),
+    DOG("mob.wolf.bark", listOf("bark", "arf", "woof")),
+    SHEEP("mob.sheep.say", listOf("baa", "baah", "baaa", "baaaa", "baaaaa")),
+    COW("mob.cow.say", listOf("moo", "mooo", "moooo")),
```

Comment:
I think using just "moo" instead of "mooo" and "moooo" should be fine?
I think it should trigger?
<img width="590" alt="image" src="https://github.com/user-attachments/assets/403e610b-245d-4591-93c5-7ffdcc65b4ab">

Reply:
You are missing these dollar signs,
![image](https://github.com/user-attachments/assets/0eea661a-51c8-4ed3-af24-eb1afde8d550)

so it will not match
![image](https://github.com/user-attachments/assets/96e399bf-f560-46dc-9c91-9bf50a80c465)

Reply:
ah okay nvm

Reply:
I changed the pattern to be more dynamic.